src/edu/psu/sweng500/team8/gui/BoardGUI.java

```diff
@@ -0,0 +1,261 @@
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][0], gridBagConstraints);
+
+		blocks[0][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][1], gridBagConstraints);
+
+		blocks[0][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 0;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 0);
+		add(blocks[0][2], gridBagConstraints);
+
+		blocks[1][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 0, 0);
+		add(blocks[1][0], gridBagConstraints);
+
+		blocks[1][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+		add(blocks[1][1], gridBagConstraints);
+
+		blocks[1][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 1;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 0, 1);
+		add(blocks[1][2], gridBagConstraints);
+
+		blocks[2][0] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 0;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 1, 1, 0);
+		add(blocks[2][0], gridBagConstraints);
+
+		blocks[2][1] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 1;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 0);
+		add(blocks[2][1], gridBagConstraints);
+
+		blocks[2][2] = new BlockGUI();
+		gridBagConstraints = new java.awt.GridBagConstraints();
+		gridBagConstraints.gridx = 2;
+		gridBagConstraints.gridy = 2;
+		gridBagConstraints.anchor = java.awt.GridBagConstraints.NORTHWEST;
+//		gridBagConstraints.insets = new java.awt.Insets(0, 0, 1, 1);
+		add(blocks[2][2], gridBagConstraints);
+	}
+
+	public void populatePanel(GameSession gameSession, boolean isRefresh) {
+
+		this.gameSession = gameSession;
+		this.selectedCell = null;
+
+		blocks[0][0].populate(gameSession.getGameBoard().getBlock(0),

```

Comment:
Assume getGameBoard is a slow call (e.g. as if it has to calculate something). In reality it's just returning the variable so it won't make much of a difference, but it's still pushing/popping the call stack unnecessarily. Thus it is more efficient to only do it once. We're talking a few nanoseconds difference, but it's a good habit to get into.